### PR TITLE
Validate topology of VkPipelineInputAssemblyStateCreateInfo

### PR DESCRIPTION
**Pull request by locke-lunarg**
[Issue #744 ](https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/744)

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -8302,6 +8302,119 @@ TEST_F(VkLayerTest, PointSizeFailure) {
     m_errorMonitor->VerifyFound();
 }
 
+TEST_F(VkLayerTest, InvaildTopolog) {
```
Comment (mark-lunarg) [inline tests/layer_validation_tests.cpp]: Invaild->Invalid.  Also, Topolog everywhere should probalby be 'Topology'

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -1458,13 +1458,42 @@ bool CoreChecks::ValidatePipelineUnlocked(std::vector<std::unique_ptr<PIPELINE_S
...
+            }
+        }
+
+        if ((pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_POINT_LIST ||
+             pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_LINE_LIST ||
+             pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST ||
+             pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_LINE_LIST_WITH_ADJACENCY ||
+             pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST_WITH_ADJACENCY ||
+             pPipeline->graphicsPipelineCI.pInputAssemblyState->topology == VK_PRIMITIVE_TOPOLOGY_PATCH_LIST) &&
+            (pPipeline->graphicsPipelineCI.pInputAssemblyState->primitiveRestartEnable == VK_TRUE)) {
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: Let's pretend I'm Tobin -- why don't we move the && parts of these big if statements to the top, maybe we get an early out? 

So, in **this** case, the primitiveRestartEnable clause would go first.  Anybody know if the compiler does this for us?

**Comment by mark-lunarg** [comment]
Added coupla comments.

File: layers/core_validation.cpp (hunk 2 above)
Comment (mikes-lunarg) [reply layers/core_validation.cpp]: The compiler can't do this for us because left-to-right evaluation and short-circuit are both mandated by the c++ spec

**Pull request merged**